File: 2015/ruby/day2/part1.rb
#!/usr/bin/ruby

begin
    File.open('input.txt', 'r') do |file|
        area = 0

        file.each_line do |line|
            dimensions = line.split('x')

            width = dimensions[0].to_i
            height = dimensions[1].to_i
            length = dimensions[2].to_i

            side1 = width * height
            side2 = width * length
            side3 = height * length

            area += 2 * (side1 + side2 + side3) + [side1, side2, side3].min
        end

        puts area
    end
rescue SystemCallError
    puts 'input.txt not found.'
end

File: 2016/python/day3/part1.py
#!/usr/bin/python

with open('input.txt') as file_handler:
    content = file_handler.readlines()

possible = 0

for line in content:
    sides = [int(side) for side in line.split()]

    if max(sides) < sum(sides) / 2.0:
        possible += 1

print possible
